feat(config): Make database backend configurable (#143) (#203)
adorsys/status-list-server: docs/database-backends.md

# Database Backend Guidance

> The server now validates the configured database backend at startup and supports the following runtime choices:

Reviewer: ```suggestion
The server validates the configured database backend at startup and supports the following runtime choices:
```

Author: Applied in 7898bf1.

- `postgres`
- `mysql`
- `sqlite`

> MariaDB is not a separate backend in this codebase. If you run MariaDB-compatible infrastructure, configure the app with `APP_DATABASE__BACKEND=mysql` and a `mysql://` URL.

Reviewer: ```suggestion
**Note**: If you run MariaDB-compatible infrastructure, configure the app with `APP_DATABASE__BACKEND=mysql` and a `mariadb://` URL.
```

Author: Not applied. sea-orm's MySQL connector only recognises the `mysql://` scheme (it checks `Url::scheme() == \"mysql\"`), so a `mariadb://` URL would pass our validation but fail at `Database::connect` with "no supporting driver". MariaDB must use `mysql://`; documented in `docs/database-backends.md`.
